### PR TITLE
feat: organised the mapsheet api

### PR DESCRIPTION
**Pull request by blacha**
This greatly increases the docmentation for all of the topographic 1:50k tile index api operations

### CODE REVIEW

File: src/utils/mapsheet.ts [hunk 1]
```diff
@@ -19,89 +47,128 @@ export interface Size {
...
+
+    const out: MapTileIndex = {
+      mapSheet: match[1],
+      gridSize: Number(match[2]),
+      x: -1,
+      y: -1,
+      name: match[0],
+      bounds: { x: 0, y: 0, width: 0, height: 0 },
+    };
+    // 1:500 has X/Y is 3 digits not 2
```
Comment (paulfouquet) [inline src/utils/mapsheet.ts:101]:
```suggestion
    // 1:500 X/Y is 3 digits not 2
```

File: src/utils/mapsheet.ts [hunk 2]
```diff
@@ -19,89 +47,128 @@ export interface Size {
...
+ * - https://data.linz.govt.nz/layer/106965-nz-1500-tile-index/ 1:500
+ **/
 export const MapSheet = {
+  /** Width of Topo 1:50k mapsheets in meters */
   height: 36_000,
+  /** Height of Topo 1:50k mapsheets in meters */
   width: 24_000,
+  /** Base scale Topo 1:50k mapsheets in meters*/
   scale: 50_000,
+  /** Map Sheets start at AS and end at CK */
```
Comment (paulfouquet) [inline src/utils/mapsheet.ts:74]:
```suggestion
  /** Map Sheets start at AS (https://www.linz.govt.nz/sites/default/files/north-island.jpg) and end at CK (https://www.linz.govt.nz/sites/default/files/south-island.jpg) */
```

Comment (blacha) [reply src/utils/mapsheet.ts:74]: Is there a  page these came from, I hesitant to linked directly to random images

Comment (paulfouquet) [reply src/utils/mapsheet.ts:74]: https://www.linz.govt.nz/products-services/maps/new-zealand-topographic-maps/topo50-map-chooser/topo50-sheet-index

Comment (blacha) [reply src/utils/mapsheet.ts:74]: this is a good link! we should link to it somewhere in this file.

**Review by paulfouquet — APPROVED** [review]

**Review by paulfouquet — APPROVED** [review]

**Pull request merged**